pkg/compact/compact.go

```diff
@@ -101,6 +104,19 @@ func newSyncerMetrics(reg prometheus.Registerer) *syncerMetrics {
 		},
 	})
 
+	m.indexCacheTraverse = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "thanos_compact_index_cache_traverse_total",
+		Help: "Total number of index cache operations.",
+	})
+	m.indexCacheBlocks = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
It think this should be always equal blocks in system, right? I would say gauge of blocks with missing index cache files would be more useful? (:

Reply:
reconsider to process index cache once if no errors happened during the process. It's expected to be a temporary decision, so I guess it's ok.